kyo-reactive-streams/shared/src/test/scala/kyo/interop/flow/SreamSubscriberTest.scala

```diff
@@ -0,0 +1,86 @@
+package kyo.interop.flow
+
+import StreamSubscriber.EmitStrategy
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Flow.*
+import kyo.*
+import kyo.Maybe.*
+
+final class JavaSubscription(subscriber: Subscriber[? >: Int], batchSize: Int, counter: java.util.concurrent.atomic.AtomicInteger)
+    extends Subscription:
+
+    private val isCanceled   = java.util.concurrent.atomic.AtomicBoolean(false)
+    private val requestCount = java.util.concurrent.atomic.AtomicInteger(0)
+
+    override def request(n: Long): Unit =
+        def loop(): Runnable =
+            new Runnable:
+                override def run(): Unit =
+                    if !isCanceled.get() && (requestCount.getAndIncrement() < n) then
+                        subscriber.onNext(counter.getAndIncrement())
+                        run()
+        requestCount.set(0)
+        discard(CompletableFuture.allOf(
+            List.fill(batchSize)(CompletableFuture.runAsync(loop()))*
+        ).get())
+    end request
+
+    override def cancel(): Unit = isCanceled.set(true)
+end JavaSubscription
+
+final class SreamSubscriberTest extends Test:
```

Comment:
I decided to test `StreamSubscriber` part only, as it's also the only part that use shared atomic state